storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -4265,7 +3973,7 @@ void xtrabackup_backup_func(void) {
 
   recv_is_making_a_backup = true;
   bool data_copying_error = false;
-  std::shared_ptr<xb::dd_tablespaces> xb_dd_space;
+  std::shared_ptr<xb::backup::dd_space_ids> xb_dd_spaces;
```

Comment:
Right, I had the same question to @rahulmalik87
I think he told me that it is required because the structure is passed to the parallel copy threads.

But after analyzing it deep, it seems that we use it at the time of populating the iterator and not after that.
So it is not used by copy threads.

@rahulmalik87 can you please check and confirm. We can make this unique_ptr or not?

Reply:
I am ok with a JIRA